apm-agent-plugins/apm-rabbitmq/apm-rabbitmq-spring/src/main/java/co/elastic/apm/agent/rabbitmq/SpringAmqpBatchMessageListenerInstrumentation.java

```diff
@@ -47,21 +57,73 @@ public String getAdviceClassName() {
     }
 
     public static class MessageListenerContainerWrappingAdvice extends BaseAdvice {
-        protected static final MessageBatchHelper messageBatchHelper;
+        private static final Logger oneTimeTransactionCreationWarningLogger;
+        private static final MessageBatchHelper messageBatchHelper;
+        private static final MessagingConfiguration messagingConfiguration;
 
         static {
             ElasticApmTracer elasticApmTracer = GlobalTracer.requireTracerImpl();
             messageBatchHelper = new MessageBatchHelper(elasticApmTracer, transactionHelper);
+            messagingConfiguration = elasticApmTracer.getConfig(MessagingConfiguration.class);
+            oneTimeTransactionCreationWarningLogger = LoggerUtils.logOnce(LoggerFactory.getLogger("Spring-AMQP-Batch-Logger"));
         }
 
-        @Nullable
-        @Advice.AssignReturned.ToArguments(@ToArgument(0))
+        @Advice.AssignReturned.ToArguments(@ToArgument(index = 0, value = 0, typing = DYNAMIC))
         @Advice.OnMethodEnter(inline = false)
-        public static List<Message> beforeOnBatch(@Advice.Argument(0) @Nullable final List<Message> messageBatch) {
-            if (!tracer.isRunning() || tracer.currentTransaction() != null || messageBatch == null) {
-                return messageBatch;
+        public static Object[] beforeOnBatch(@Advice.This Object thiz,
+                                             @Advice.Argument(0) @Nullable final List<Message> messageBatch) {
+
+            List<Message> processedBatch = messageBatch;
+            Transaction batchTransaction = null;
+
+            if (tracer.isRunning() && messageBatch != null && !messageBatch.isEmpty()) {
```

Comment:
[minor] maybe return early in case where tracer is not running or batch is empty to prevent some if nesting